BluetoothAdvertisementsKotlin/app/src/main/java/com/example/bluetoothadvertisements/adapter/ScannerAdapter.kt

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bluetoothadvertisements.adapter
+
+import android.bluetooth.le.ScanResult
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.bluetoothadvertisements.databinding.VhScanResultBinding
+
+private const val TAG = "ScannerAdapter"
+
+class ScannerAdapter : RecyclerView.Adapter<ScannerAdapter.ScanResultVh>() {
+
+    private var itemsList: MutableList<ScanResult> = arrayListOf()
+
+    fun setItems(mutableList: MutableList<ScanResult>) {
+        if (mutableList != itemsList) {
+            itemsList = mutableList
+            notifyDataSetChanged()
+        }
+    }
+
+    fun addSingleItem(item: ScanResult) {
+        /**
+         * In this particular case the data coming in may be duplicate. So check that only unique
+         * elements are admitted: the device Id + device name should create a unique pair.
+         * removeIf requires API level 24, so using removeAll here. But feel free to use any of
```

Comment:
The method removeIf() ... removeAll()

Reply:
I don't follow. Are you saying not to mention "removeIf()" in the comments? or to change min sdk from 21 to 24?

Reply:
Never mind, I hadn't used removeIf before from Java. TIL.. ha ha, you can ignore this comment.